src/session/request-event/classes.js

```diff
@@ -0,0 +1,47 @@
+export class RequestEvent {
```

Comment:
At first it seems it relates to `request-pipeline`, why we placed in in the `session/` dir?
And I guess now it's not the best naming here. Actually `request-event/classes.js` doesn't say anything. 
Maybe something like:
```
-- request-pipeline/events
  | -- request-event
  | -- configure-response-event-options
  | -- configure-response-event
  | -- response-event
  | -- events
  | -- info
```

?

Reply:
> At first it seems it relates to request-pipeline

No. `Request pipeline` is all about request stages (build request params, send to destination server and etc) and respond strategy for various response types: xhr, websocket, special page (`about:blank`) and etc.

Events (in my case callbacks on request rule filters) raised per session and connected with session. 
Session has some methods related to event: https://github.com/DevExpress/testcafe-hammerhead/pull/1527/files#diff-fbb4b02e77cf63674193a3e96d8eadfdR156

> Actually request-event/classes.js doesn't say anything.

Agree. I split `classes.js` on several files.